src/crypto_backend.rs

```diff
@@ -18,13 +23,24 @@ pub trait LocalKeyPair {
     /// Export the public key component in the
     /// binary uncompressed point representation.
     fn pub_as_raw(&self) -> Result<Vec<u8>>;
+    /// Import a keypair from its raw components.
+    fn from_raw_components(components: &EcKeyComponents) -> Result<Self>
+    where
+        Self: Sized;
+    /// Export the raw components of the keypair as follows:
+    /// EC parameters `d`, `x` then `y`.
+    fn raw_components(&self) -> Result<(EcKeyComponents)>;
+}
+
+pub struct EcKeyComponents {
+    pub d: Vec<u8>,
```

Comment:
Instead of these pub fields, let's expose methods so the internal representation of `EcKeyComponents` can change.

Reply:
Also is this struct name right?